src/lib.rs

```diff
@@ -306,3 +311,44 @@ fn is_multiline_start(val: &str) -> Option<QuoteType> {
     }
     None
 }
+
+/// Prints information about the new version to `STDOUT` if a new version available
+fn print_new_version_if_available(args: &clap::ArgMatches) {
+    if args.is_present("not-check-updates") || args.is_present("quiet") {
+        return;
+    }
+
+    #[cfg(not(feature = "stub_check_version"))]
+    let current_version = env!("CARGO_PKG_VERSION");
+
+    #[cfg(feature = "stub_check_version")]
+    let current_version = "3.0.0";
+
+    let pkg_name = env!("CARGO_PKG_NAME");
+    let interval = Duration::from_secs(60 * 60 * 24);
+
+    #[cfg(not(feature = "stub_check_version"))]
```

Comment:
Honestly, call different functions depends on feature - looks a bit weird. I see high coupling in this external functions.
It seems like this only needed for test purposes. 

I'd prefer to have api with different instances/structs of `UpdateInformer` - but with single interface/trait
Like this:

```rust
#[cfg(not(feature = "stub_check_version"))]
let update_informer = UpdateInformer::new(Crates, interval); // it seems that `pkg_name` and `version` could be moved to constructor (fn new), because `env!` macro will be expanded on compile time

#[cfg(feature = "stub_check_version")]
let update_informer = StubUpdateInformer::new(Crates, interval, "3.1.1");

let result = update_informer.check_version(); // check_version - fn from `Trait` implemented in different ways for Stub/non-stub
```

Is it possible to make similar changes in the dependency part? 
If you disagree, it's okay, let's leave it this way

Reply:
Thank you for your feedback!
I like your option, but I'll do it a bit later.